src/addresstype.h

```diff
@@ -69,22 +69,25 @@ struct WitnessV1Taproot : public XOnlyPubKey
 //! CTxDestination subtype to encode any future Witness version
 struct WitnessUnknown
 {
-    unsigned int version;
-    unsigned int length;
-    unsigned char program[40];
```

Comment:
AFAICT, changing this to `std::vector` was causing the `-Wuninitialized` warning. I played around with overriding the default constructor but kept getting the same error. What was causing the error was this line:

https://github.com/bitcoin/bitcoin/blob/e3d6bd41629a0735d7fb1475b77ffd3be7280c9a/src/util/message.cpp#L50

where we temporarily create a `CTxDestination(PKHash(pubkey))` to compare it to a `CTxDestination`. From what I can tell, creating the temporary `CTxDestination` with only a `PKHash` in it and then immediately calling the destructor was triggering the error for `WitnessUnknown` since it was never really constructed. Either way, I think it makes more sense to just create a `PKHash` and then compare it to the `CTxDestination` with a `std::get_if` like so: 

```
-    if (!(CTxDestination(PKHash(pubkey)) == destination)) {
+    if (!(PKHash(pubkey) == *std::get_if<PKHash>(&destination))) {
```

I made this change in https://github.com/bitcoin/bitcoin/commit/50fd2a5455288a87249ef5ed73f7db34f814bf8a and verified that everything compiles without the warning, feel free to cherry-pick it

Reply:
Done as suggested.

Reply:
in https://github.com/bitcoin/bitcoin/pull/28246/commits/c14ad74f8eeaa744d831a77654164c856a86fdee:

I'm getting the `-Wunitialized` error again, due to this line: 

https://github.com/bitcoin/bitcoin/blob/c14ad74f8eeaa744d831a77654164c856a86fdee/src/test/fuzz/key.cpp#L189

IIRC, this was fixed in a previous version of this PR, not sure when it changed, but this should be:

```
-        assert(CTxDestination{PKHash{pubkey}} == tx_destination);
+        assert(PKHash{pubkey} == *std::get_if<PKHash>(&tx_destination));
```

Reply:
Fixed in fuzz/key.cpp